nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/FetchHBaseRow.java

```diff
@@ -0,0 +1,407 @@
+     */
+    private interface FetchHBaseRowHandler extends ResultHandler {
+
+        /**
+         * @return returns the flow file reference that was used by this handler
+         */
+        FlowFile getFlowFile();
+
+        /**
+         * @return returns true if this handler handled a row
+         */
+        boolean handledRow();
+
+    }
+
+    /**
+     * A FetchHBaseRowHandler that writes the resulting row to the FlowFile content.
+     */
+    private static class FlowFileContentHandler implements FetchHBaseRowHandler {
+
+        private FlowFile flowFile;
+        private final ProcessSession session;
+        private final RowSerializer serializer;
+        private boolean handledRow = false;
+
+        public FlowFileContentHandler(final FlowFile flowFile, final ProcessSession session, final RowSerializer serializer) {
+            this.flowFile = flowFile;
+            this.session = session;
+            this.serializer = serializer;
+        }
+
+        @Override
+        public void handle(byte[] row, ResultCell[] resultCells) {
+            flowFile = session.write(flowFile, (out) -> {
+                serializer.serialize(row, resultCells, out);
+            });
+            handledRow = true;
+        }
+
+        @Override
+        public FlowFile getFlowFile() {
+            return flowFile;
+        }
+
+        @Override
+        public boolean handledRow() {
+            return handledRow;
+        }
+    }
+
+    /**
+     * A FetchHBaseRowHandler that writes the resulting row to FlowFile attributes.
+     */
+    private static class FlowFileAttributeHandler implements FetchHBaseRowHandler {
+
+        private FlowFile flowFile;
+        private final ProcessSession session;
+        private final Charset charset;
+        private final boolean base64EncodeValues;
+        private boolean handledRow = false;
+
+        public FlowFileAttributeHandler(final FlowFile flowFile, final ProcessSession session, final Charset charset, final boolean base64EncodeValues) {
+            this.flowFile = flowFile;
+            this.session = session;
+            this.charset = charset;
+            this.base64EncodeValues = base64EncodeValues;
+        }
+
+        @Override
+        public void handle(byte[] row, ResultCell[] resultCells) {
+            final int size = resultCells == null ? 0 : resultCells.length;
+            final Map<String,String> attributes = new HashMap<>(size);
+
+            for (final ResultCell cell : resultCells) {
+                String cellFamily = new String(cell.getFamilyArray(), cell.getFamilyOffset(), cell.getFamilyLength(), charset);
+                String cellQualifier = new String(cell.getQualifierArray(), cell.getQualifierOffset(), cell.getQualifierLength(), charset);
+                String cellValue = ResultCellUtil.getCellValue(cell, charset, base64EncodeValues);
+                attributes.put(cellFamily + ":" + cellQualifier, cellValue);
```

Comment:
In creating the map key, you end up creating 4 Strings (maybe 5?). If you can find a way, appending the data as bytes into one array and then encoding it as UTF-8 once in a String would save a few String allocs.